### PR TITLE
Implement ssl.curves.list and ssl.sigalgs.list configuration settings

### PR DESCRIPTION
**Pull request by jvgutierrez**
At WMF we are conducting a [TLS security review of our kafka stack](https://phabricator.wikimedia.org/T182993). As part of this review we identified that we need to:
 * Restrict the Eliptic curves (`ssl.curves.list`) being used.
 * Restrict the certificate signature algorithms (`ssl.sigalgs.list`) being used.

Both settings can be controlled using the following OpenSSL API functions introduced in OpenSSL 1.0.2:
* `SSL_CTX_set1_curves_list(3)`
* `SSL_CTX_set1_sigalgs_list(3)`

I've successfully tested the submitted code using the provided `rdkafka_example` with the following parameters:
```
examples/rdkafka_example -P -b localhost:4433 -t test -d SECURITY -X security.protocol=SSL -X ssl.cipher.suites=ECDHE-ECDSA-AES256-GCM-SHA384 -X ssl.ca.location=ca/ca.pem -X ssl.certificate.location=ca/prototype-client.pem -X ssl.key.location=ca/prototype-client-key.pem -X ssl.curves.list=P-256 -X ssl.sigalgs.list=ECDSA+SHA256
```

On the server side I've used `openssl s_server`. Here is the output from the server without and with the new settings:

without ssl.curves.list and without ssl.sigalgs.list:
```
Signature Algorithms: RSA+SHA512:DSA+SHA512:ECDSA+SHA512:RSA+SHA384:DSA+SHA384:ECDSA+SHA384:RSA+SHA256:DSA+SHA256:ECDSA+SHA256:RSA+SHA224:DSA+SHA224:ECDSA+SHA224:RSA+SHA1:DSA+SHA1:ECDSA+SHA1
Supported Elliptic Curves: X25519:P-256:P-521:P-384
```
with `-X ssl.curves.list=P-256 -X ssl.sigalgs.list=ECDSA+SHA256`:
```
Shared Signature Algorithms: ECDSA+SHA256
Supported Elliptic Curves: P-256
```

### CODE REVIEW

File: src/rdkafka_transport.c [hunk 1]
```diff
@@ -836,6 +836,33 @@ int rd_kafka_transport_ssl_ctx_init (rd_kafka_t *rk,
...
 
+#if OPENSSL_VERSION_NUMBER >= 0x1000200fL
+	/* Curves */
+	if (rk->rk_conf.ssl.curves_list) {
+		rd_kafka_dbg(rk, SECURITY, "SSL",
+				 "Setting curves list: %s",
+				 rk->rk_conf.ssl.curves_list);
+		if (!SSL_CTX_set1_curves_list(ctx,
+						rk->rk_conf.ssl.curves_list)) {
+							rd_snprintf(errstr, errstr_size,
```
Comment (edenhill) [inline src/rdkafka_transport.c]: indent looks off here. Use linux c-style.

File: src/rdkafka_transport.c [hunk 2]
```diff
@@ -836,6 +836,33 @@ int rd_kafka_transport_ssl_ctx_init (rd_kafka_t *rk,
...
+		}
+	}
+
+	/* Certificate signature algorithms */
+	if (rk->rk_conf.ssl.sigalgs_list) {
+		rd_kafka_dbg(rk, SECURITY, "SSL",
+				 "Setting signature algorithms list: %s",
+				 rk->rk_conf.ssl.sigalgs_list);
+		if (!SSL_CTX_set1_sigalgs_list(ctx,
+						rk->rk_conf.ssl.sigalgs_list)) {
```
Comment (edenhill) [inline src/rdkafka_transport.c]: dito indent

**Comment by jvgutierrez** [comment]
Oops! sorry about that, hopefully I got it right this time :)

**Comment by jvgutierrez** [comment]
@edenhill anything else to move this PR forward?

**Comment by edenhill** [comment]
I'm happy with this but there's been some discussion whether changes like this should go through a KIP, or at least kafka-dev mailing list discussion, to make sure the client configuration properties are synced between librdkafka and Java.

File: src/rdkafka_conf.c [hunk 3]
```diff
@@ -470,6 +470,22 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
 	  "protocol. See manual page for `ciphers(1)` and "
 	  "`SSL_CTX_set_cipher_list(3)."
 	},
+#if OPENSSL_VERSION_NUMBER >= 0x1000200fL
+	{ _RK_GLOBAL, "ssl.curves.list", _RK_C_STR,
```
Comment (edenhill) [inline src/rdkafka_conf.c]: please use 8 whitespaces for indent (I know this is inconsistent in the code for historical reasons, but all new code should use this new style)

Comment (jvgutierrez) [reply src/rdkafka_conf.c]: no problem, fixed

**Comment by jvgutierrez** [comment]
Awesome! nice to see this PR moving forward :)

**Comment by edenhill** [comment]
Thanks for this!

**Pull request merged**